network/src/noise/handshake.rs

```diff
@@ -279,12 +316,21 @@ impl NoiseUpgrader {
         mut socket: TSocket,
     ) -> io::Result<(NoiseStream<TSocket>, PeerId)>
     where
-        TSocket: AsyncRead + AsyncWrite + Unpin,
+        TSocket: AsyncRead + AsyncWrite + Debug + Unpin,
     {
         // buffer to contain the client first message
         let mut client_message = [0; Self::CLIENT_MESSAGE_SIZE];
 
         // receive the prologue + first noise handshake message
+        sl_info!(
+            network_log(network_events::NOISE_UPGRADE, &self.network_context)
+                .data("origin", "server")
+                .data("step", "reading_data")
+                .log(format!(
```

Comment:
should log take in a parameterized string like format?

Reply:
why log though? Why not `data_display()`?

Reply:
Yeah for simplicity getting out the door, I didn't.  It is annoying that you have to do `format!` on each one.  Those format strings are mostly handled most easily by macros.

`log` is the replacement for a context text log line like you would use in `info!("Some info: {}", info)`. In this example it is `Some info: info`.  The objective was to provide a way to migrate all logs to structured logs from text logs.

Reply:
@mimoo, @gregnazario 's second paragraph is a response to your question.

Reply:
isn't greg pointing out that you shouldn't use log :D ?

Reply:
no, the point is that if you want your log to show up in text logs you need to do both... do I want a second line that is just "info!(...)" that would then route to structured log and then have two structured logs or use the model in the code that will ensure that text and structured are logged only once.